src/popup/components/PayWebsiteForm.tsx

```diff
@@ -42,54 +34,72 @@ export const PayWebsiteForm = () => {
 
   const form = React.useRef<HTMLFormElement>(null);
   const [isSubmitting, setIsSubmitting] = React.useState(false);
-  const [buttonState, setButtonState] =
-    React.useState<keyof typeof BUTTON_STATE>('idle');
-  const isIdle = React.useMemo(() => buttonState === 'idle', [buttonState]);
+  const [msg, setMsg] = React.useState<null | {
+    type: 'success' | 'warn';
+    message: string;
+  }>(null);
```

Comment:
Maybe this is going to be more explicit?
```suggestion
  const [payStatus, setPayStatus] = React.useState<null | {
    type: 'full' | 'partial';
    message: string;
  }>(null);
```